src/pydata_sphinx_theme/assets/styles/sections/_header.scss

```diff
@@ -82,8 +82,6 @@
 
   // Contains the navigation links within the navbar
   .navbar-nav {
-    display: flex;
```

Comment:
Oh but the _navbar-nav.scss file changes the `ul` to `display: block`... Hmm, this might be more complicated than I realize. I'm going to put this PR into draft mode.